codecparsers/mpeg2_parser.h

```diff
@@ -378,23 +378,23 @@ namespace MPEG2 {
 
         inline bool bitReaderReadMarker(bool value) const
         {
-            bool peekMarker;
+            bool readMarker, ret;
             // peek 1 bit marker
-            peekMarker = static_cast<bool>(m_bitReader->read(1));
-            return peekMarker == value;
+            ret = (m_bitReader->readT(readMarker));
+            return readMarker == value && ret;
         }
 
         inline bool bitReaderPeekMarker(bool value) const
         {
-            bool readMarker;
+            bool peekMarker;
             // read 1 bit marker
-            readMarker = static_cast<bool>(m_bitReader->peek(1));
-            return readMarker == value;
+            peekMarker = static_cast<bool>(m_bitReader->peek(1));
```

Comment:
As you commented on pull request #524 it will require to change peek function.  This can be fixed on a later patch as modifying peek will affect vc1 as well.